### PR TITLE
Add train_instruct_pix2pix_save_best.py that saves the model with minimum training loss

### PR DESCRIPTION
**Pull request by Lliillyy**
# What does this PR do?

Add `train_instruct_pix2pix_save_best.py` script that saves the model with minimum training loss during the training process. Tracked `min_train_loss` and `min_global_step` during training.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

- Training examples: @sayakpaul and @patrickvonplaten

### CODE REVIEW

**Comment by sayakpaul** [comment]
Thank you for the contribution but we cannot have too specific scripts. Our training scripts are not exhaustive and one of the main purposes is to allow others to build on top of them.

**Comment by sayakpaul** [comment]
I am going to close the PR. But I deeply appreciate your willingness to contribute. I hope we find more contribution opportunities for you in the future.
